lib/membrane/video_compositor/handler.ex

```diff
@@ -0,0 +1,75 @@
+defmodule Membrane.VideoCompositor.Handler do
+  @moduledoc """
+  Module defining behaviour of handlers.
+
+  Implementing handler allows to provide custom implementation and
+  react to various events, among others by setting a new scene and/or state.
+  """
+
+  alias __MODULE__.CallbackContext
+  alias Membrane.{Pad, StreamFormat, Time}
+  alias Membrane.VideoCompositor.Scene
+
+  @typedoc """
+  Type of user-managed inner state of the handler.
+  """
+  @type state :: any()
+
+  @type video_details :: {pad :: Pad.ref_t(), format :: StreamFormat.t()}
+
+  @typedoc """
+  Type of a valid return value from a callback.
+  """
+  @type callback_return :: immediate_callback_return() | timed_callback_return()
+
+  @typedoc """
+  Type of a valid return value from most of the callbacks.
```

Comment:
I would still specify, that in this case the scene is changed immediately, rather than saying that it's common for most callbacks